libnmstate/nm/route.py

```diff
@@ -123,3 +145,32 @@ def _nm_route_to_route(nm_route, table_id, iface_name):
         Route.NEXT_HOP_ADDRESS: next_hop,
         Route.METRIC: metric,
     }
+
+
+def add_routes(setting_ip, routes):
+    default_gateway_defined = False
+    for route in routes:
+        if route[Route.DESTINATION] in (IPV4_DEFAULT_GATEWAY_DESTINATION,
+                                        IPV6_DEFAULT_GATEWAY_DESTINATION):
+            if default_gateway_defined:
+                raise NmstateNotImplementedError(
+                    'Only a single default gateway is supported.')
+            default_gateway_defined = True
+            setting_ip.props.gateway = route[Route.NEXT_HOP_ADDRESS]
+            if route[Route.TABLE_ID] != NM_ROUTE_TABLE_DEFAULT:
+                setting_ip.props.route_table = route[Route.TABLE_ID]
+            if route[Route.METRIC] != NM_ROUTE_DEFAULT_METRIC:
+                setting_ip.props.route_metric = route[Route.METRIC]
+            continue
+
+        dst, prefix_len = route[Route.DESTINATION].split('/')
+        prefix_len = int(prefix_len)
+        family = socket.AF_INET6 if iplib.is_ipv6_address(dst) \
+            else socket.AF_INET
+        metric = route.get(Route.METRIC, NM_ROUTE_DEFAULT_METRIC)
+        next_hop = route[Route.NEXT_HOP_ADDRESS]
+        rt = nmclient.NM.IPRoute.new(family, dst, prefix_len, next_hop, metric)
+        if route[Route.TABLE_ID] != NM_ROUTE_TABLE_DEFAULT:
+            rt.set_attribute(NM_ROUTE_TABLE_ATTRIBUTE,
+                             GLib.Variant.new_uint32(route[Route.TABLE_ID]))
+        setting_ip.add_route(rt)
```

Comment:
The `setting_ip` is short of `NM.SettingIPConfig`.
I believe the function name `add_routes()` already indicate it will make changes to the `setting_ip`.